pkg/openstack/instances.go

```diff
@@ -686,6 +716,51 @@ func nodeAddresses(srv *servers.Server, ports []ports.Port, networkingOpts Netwo
 	return addrs, nil
 }
 
+// getSubports returns subports attached to the node via trunk
+func getSubports(client *gophercloud.ServiceClient, serverID string) ([]ports.Port, error) {
+	listOpts := ports.ListOpts{
+		DeviceID:    serverID,
+		DeviceOwner: "compute:nova",
+	}
+
+	type portWithTrunkDetails struct {
+		ports.Port
+		trunk_details.TrunkDetailsExt
+	}
+
+	var subports []ports.Port
+	var allPortDetails []portWithTrunkDetails
+
+	allPages, err := ports.List(client, listOpts).AllPages()
+	if err != nil {
+		return subports, err
+	}
```

Comment:
We can't do this: it's adding the second port list for everybody. We've already done this query. We need to re-use the existing ports query.

Reply:
ok, same as above, rebase results in a new PR, will take care of the comments in the new PR:
https://github.com/kubernetes/cloud-provider-openstack/pull/2306